src/concurrency/weak_memory.rs

```diff
@@ -6,7 +6,7 @@
 //! but it is incapable of producing all possible weak behaviours allowed by the model. There are
 //! certain weak behaviours observable on real hardware but not while using this.
 //!
-//! Note that this implementation does not take into account of C++20's memory model revision to SC accesses
+//! Note that this implementation does not fully take into account of C++20's memory model revision to SC accesses
```

Comment:
Please describe somewhere in this comment how what this file does differs from the paper to account partially for the C++20 changes.